Creates CI/CD pipeline and project configuration with Poetry

lucassouto opened the pull request:
## Add

- Project configuration with Poetry
- Tests for create and patch secrets methods
- Gitignore file
- Publishing the package through the pipeline

Commit:
chore(nameko-vault): Add pyproject.toml

Commit:
chore(nameko-vault): Remove setup.py

Commit:
chore(CI): Add github actions

Commit:
style(black): Apply black format

Commit:
chore(CI): Publish package in CI

Commit:
tests(provider): Add tests create and patch method

Commit:
chore(CI): Add test step

oesgalha commented on pyproject.toml:
```diff
@@ -0,0 +1,57 @@
+[tool.poetry]
+name = "nameko-vault"
+version = "0.1.2"
```
Since you are adding new features, the next version should be 0.2.0 instead of 0.1.2

oesgalha commented on pyproject.toml:
```diff
@@ -0,0 +1,57 @@
+name = "nameko-vault"
+version = "0.1.2"
+description = "A Nameko extension to provide connection with Vault"
+authors = ["Instruct Developers <contato@instruct.com.br>"]
```
Now we have an exclusive mail address for library related contact :raised_hands: 
Can you change the mail to "oss `at` instruct `dot` com `dot` br?

oesgalha commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,43 @@
+      run: |
+        python -m pip install poetry
+        poetry install
+    - name: Lint check
+      run: poetry run black --check .
```
The lint step does not need to run against 3 versions of Python, just the latest is fine.

Note: this can be addressed in another PR

lucassouto replied:
OK this one I will leave for a future PR

oesgalha commented on poetry.lock:
```diff
@@ -0,0 +1,806 @@
+[[package]]
```
Since this is a library, add the `.lock` file to .gitinore.

From Poetry docs:

> For libraries it is not necessary to commit the lock file.

https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control

---

Here is another source about when you should keep lockfiles in (or out) of version control:
https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

oesgalha commented on tests/tests_provider.py:
```diff
@@ -0,0 +1,76 @@
+        'auth': None,
+    }
+
+    service.vault.create_or_update_kv_secret.return_value = mock_return
+
+    assert service.create_secret(path, secret) == mock_return
+    service.vault.create_or_update_kv_secret.assert_called_once_with(
+        path, secret
+    )
```
I think those tests are testing the example service behavior, not the `VaultProvider` behavior.

Note: this can also be addressed in another PR, the CI pipeline with tests running is valuable

lucassouto replied:
OK this one I will leave for a future PR

oesgalha reviewed (changes requested):
Excellent work! :clap: 

Added some comments.
I don't think it's required to address them all in this PR.

oesgalha commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,43 @@
+      run: poetry build
+    - name: Publish package
+      run: |
+        poetry publish -u ${{ secrets.PYPI_USER }} -p ${{ secrets.PYPI_PASS }} --no-interaction
```
We're going to publish packages with [API token](https://pypi.org/help/#apitoken):
```suggestion
        poetry publish -u "__token__" -p ${{ secrets.PYPI_TOKEN }} --no-interaction
```

Commit:
chore(package): Remove lockfile

- From Poetry docs:
`For libraries it is not necessary to commit the lock file.`

https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control

Here is another source about when you should keep lockfiles in (or out)
of version control:
https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

Commit:
chore(package): Update package version

- Change package version

- Change author email

Commit:
chore(CI): Change publishing method to use token

oesgalha reviewed (approved):
Thank you :+1:
